pkg/node/node.go

```diff
@@ -387,59 +396,8 @@ func NewBee(addr string, logger logging.Logger, o Options) (*Bee, error) {
 		b.debugAPIServer = debugAPIServer
 	}
 
-	addresses, err := addressbook.Overlays()
-	if err != nil {
-		return nil, fmt.Errorf("addressbook overlays: %w", err)
-	}
-
-	var count int32
-
-	// add the peers to topology and allow it to connect independently
-	for _, o := range addresses {
-		err = topologyDriver.AddPeer(p2pCtx, o)
-		if err != nil {
-			logger.Debugf("topology add peer from addressbook: %v", err)
-		} else {
-			count++
-		}
-	}
-
-	// Connect bootnodes if the address book is clean
-	if count == 0 {
-		var wg sync.WaitGroup
-		for _, a := range o.Bootnodes {
-			wg.Add(1)
-			go func(a string) {
-				defer wg.Done()
-				addr, err := ma.NewMultiaddr(a)
-				if err != nil {
-					logger.Debugf("multiaddress fail %s: %v", a, err)
-					logger.Warningf("connect to bootnode %s", a)
-					return
-				}
-				var count int
-				if _, err := p2p.Discover(p2pCtx, addr, func(addr ma.Multiaddr) (stop bool, err error) {
-					logger.Tracef("connecting to bootnode %s", addr)
-					_, err = p2ps.ConnectNotify(p2pCtx, addr)
-					if err != nil {
-						if !errors.Is(err, p2p.ErrAlreadyConnected) {
-							logger.Debugf("connect fail %s: %v", addr, err)
-							logger.Warningf("connect to bootnode %s", addr)
-						}
-						return false, nil
-					}
-					logger.Tracef("connected to bootnode %s", addr)
-					count++
-					// connect to max 3 bootnodes
-					return count > 3, nil
-				}); err != nil {
-					logger.Debugf("discover fail %s: %v", a, err)
-					logger.Warningf("discover to bootnode %s", a)
-					return
-				}
-			}(a)
-		}
-		wg.Wait()
+	if err := kad.Start(p2pCtx); err != nil {
```

Comment:
not sure if i understand why this is needed as a discrete step in the kademlia start up sequence? is there anything preventing the address book to populate kademlia as part of the kademlia constructor?

Reply:
I guess not, but I had a weird feeling when using it that way. Because, we need kademlia as a dependency on some other components so it is created pretty "early", while this kind of logic where there is an ongoing loop looks more like a "service" to me. To put it simple, I liked the idea of being able to instantiate other components with kademlia, but not start the logic that is driving rest of the system as well, such as connecting, notifying, etc... LMT if this makes sense?

Reply:
This does give more control to the process, indeed. Even it may seem an overhead, I think that it is a safe approach.

Reply:
that's fine, but the `manage` loop is started in the constructor. it falls under this same case, so either pull the manage goroutine start to this function, or have both in the ctor. WDYT?

Reply:
Sure, done!